code/modules/mining/lavaland/loot/tendril_loot.dm

```diff
@@ -247,7 +247,7 @@
 
 		to_chat(user, "<span class='notice'>You release the wisp. It begins to bob around your head.</span>")
 		icon_state = "lantern"
-		wisp.orbit(user, 20)
+		INVOKE_ASYNC(wisp, /atom/movable/.proc/orbit, user, 20)
```

Comment:
So the issue here was an infinite while loop in the orbit proc and the invoke async allows this proc to continue on?

Reply:
Exactly, yes.